src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let modSeparator = separator;
+
+    if (separator === null) {
+      modSeparator = 'null';
+    }
```

Comment:
you don't need this check

Reply:
in the beginning my code was three times bigger. how to learn to make it like that right away

Reply:
it's okay, you will achieve it with practice
just keep doing